src/main/java/mods/railcraft/common/blocks/wayobjects/RoutingLogic.java

```diff
@@ -133,6 +134,17 @@ private Condition parseLine(String line, Deque<Condition> stack) throws RoutingL
         } catch (Exception ex) {
             throw new RoutingLogicException("railcraft.gui.routing.logic.malformed.syntax", line);
         }
+        if (line.equals("TRUE"))
+            return new ConstantCondition(true);
+        if (line.equals("FALSE"))
+            return new ConstantCondition(false);
+        try {
+            return new ConstantCondition(Integer.parseInt(line));
+        } catch (NumberFormatException ignored) {
+            // not an integer; pass through
+        } catch (IllegalArgumentException ex) {
+            throw new RoutingLogicException("railcraft.gui.routing.logic.constant.invalid", line);
+        }
```

Comment:
That is actually what I was trying to describe in the original issue... apologies if it was unclear. Agreed that it's unintuitive, though, unless you frequently use a language where it's an idiom.